[Bug]: component.d.ts file has incorrect import causing type errors

### Version

1.41.2

### Steps to reproduce

Try to run `tsc` with a project importing `@playwright/experimental-ct-react`

![image](https://github.com/microsoft/playwright/assets/47463734/689a956d-fb7f-472d-b00c-4c8673ffa8ae)


### Expected behavior

Expected to see no type errors

### Actual behavior

The following error:

`node_modules/@playwright/experimental-ct-core/types/component.d.ts(17,37): error TS2307: Cannot find module '../src/injected/importRegistry' or its corresponding type declarations.`

### Additional context

I went back to a couple other earlier versions (1.41.1 & 1.41.0) and those had the same behavior for component testing

### Environment

```shell
System:
    OS: Linux 6.5 Debian GNU/Linux 12 (bookworm) 12 (bookworm)
    CPU: (12) x64 Intel(R) Core(TM) i7-10750H CPU @ 2.60GHz
    Memory: 18.67 GB / 31.09 GB
    Container: Yes
  Binaries:
    Node: 18.18.0 - /usr/local/bin/node
    Yarn: 1.22.21 - /usr/local/bin/yarn
    npm: 9.8.1 - /usr/local/bin/npm
  Languages:
    Bash: 5.2.15 - /usr/bin/bash
  npmPackages:
    @playwright/test: ^1.41.2 => 1.41.2
```

--- a/packages/playwright-ct-core/src/injected/importRegistry.ts
+++ b/packages/playwright-ct-core/src/injected/importRegistry.ts
@@ -47,3 +47,9 @@ export class ImportRegistry {
     return importedObject;
   }
 }
+
+declare global {
+  interface Window {
+    __pwRegistry: ImportRegistry;
+  }
+}

--- a/packages/playwright-ct-core/types/component.d.ts
+++ b/packages/playwright-ct-core/types/component.d.ts
@@ -55,7 +55,6 @@ declare global {
     __pw_hooks_after_mount?: (<HooksConfig extends JsonObject = JsonObject>(
       params: { hooksConfig?: HooksConfig; [key: string]: any }
     ) => Promise<void>)[];
-    __pwRegistry: ImportRegistry;
     // Can't start with __pw due to core reuse bindings logic for __pw*.
     __ctDispatchFunction: (ordinal: number, args: any[]) => void;
     __pwUnwrapObject: (value: any) => Promise<any>;

